torchgeometry/image/color.py

```diff
@@ -1,21 +1,178 @@
 import torch
 import torch.nn as nn
+import numpy as np
 
 
-class RgbToGrayscale(nn.Module):
-    r"""Convert image to grayscale version of image.
+class RgbToHsV(nn.Module):
 
+    """convert image from RGB to HSV.
     The image data is assumed to be in the range of (0, 1).
 
+    args:
+        image (torch.tensor): RGB image to be converted to HSV.
+
+    returns:
+        torch.tensor: HSV version of the image.
+
+    shape:
+        - image: :math:`(*, 3, h, w)`
+        - output: :math:`(*, 3, h, w)`
+
+    """
+
+    def __init__(self) -> None:
+        super(RgbToHsV, self).__init__()
+
+    def forward(self, image: torch.Tensor) -> torch.Tensor:  # type: ignore
+        return rgb_to_hsv(image)
+
+
+def rgb_to_hsv(image):
+
+    """Convert an RGB image to HSV
+
     Args:
-        input (torch.Tensor): Image to be converted to grayscale.
+        input (torch.Tensor): RGB Image to be converted to HSV.
 
     Returns:
-        torch.Tensor: Grayscale version of the image.
+        torch.Tensor: HSV version of the image.
+    """
+
+    if not torch.is_tensor(image):
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
+                         .format(image.shape))
+
+    r = image[..., 0, :, :]
```

Comment:
using typing to help functions tracing:
`r: torch.Tensor = image[..., 0, :, :]`